pkg/handlers/v1/enqueue.go

```diff
@@ -0,0 +1,34 @@
+package v1
+
+import (
+	"context"
+
+	producer "github.com/asecurityteam/component-producer"
+	"github.com/asecurityteam/ipam-facade/pkg/domain"
+	"github.com/asecurityteam/ipam-facade/pkg/logs"
+)
+
+// EnqueueHandler enqueues IPAM data sync requests
+type EnqueueHandler struct {
+	Producer              producer.Producer
+	RandomNumberGenerator domain.RandomNumberGenerator
+	LogFn                 domain.LogFn
+}
+
+// Handle creates a job ID and enqueues the sync request with that ID
+func (h *EnqueueHandler) Handle(ctx context.Context) (JobMetadata, error) {
+	jobID, err := h.RandomNumberGenerator.NewRandom()
+	if err != nil {
+		h.LogFn(ctx).Error(logs.SyncError{Reason: err.Error()})
+		return JobMetadata{}, err
+	}
+	jobMetadata := JobMetadata{JobID: jobID.String()}
+
+	_, err = h.Producer.Produce(ctx, jobMetadata)
```

Comment:
info: The left hand value returned here is a potentially modified form of the original event. The producer contract is such that it is intended to return the exact value it published in case it differs from the input. For example, the benthos implementation of the producer component could have any number of processors enabled via the YAML configuration that result in a modified body. The rules for how the output is modified are documented here: https://docs.benthos.dev/serverless/lambda/#running-without-an-output. I don't expect that our deployments will have many cases where this output differs from our input but it is possible for a 3rd party to configure event production differently than us.

I'm still slightly conflicted over the best way to handle this situation. I generally prefer to align the return of the function with the value we produced because we could, ideally, externalize the event production at some point in the future. If so, we should be able to switch our producer driver to `NULL` and let something else handle the function's return value. I also don't recommend mixing Benthos processors and custom code to perform the business logic of a function. The event handling logic should either be entirely contained within the Benthos configuration such that we could deploy Benthos Serverless directly or entirely contained within our function such that Benthos is but one of several options for emitting the results. I also don't appreciate the loss of type safety introduced by the "generic" producer interface and would prefer to not have it infect the function signature.

All of this is to say that there's an edge case here where we may lose data if a user installs an unexpected producer configuration. I'm not sure that's something we should necessarily support but I'm not 100% convinced yet that we shouldn't. No changes requested. Only some FYI.